src/Test/Runner.elm

```diff
@@ -230,22 +236,62 @@ distributeSeeds runs seed test =
             }
 
         Internal.Labeled description subTest ->
-            let
-                next =
-                    distributeSeeds runs seed subTest
-            in
-            { seed = next.seed
-            , all = List.map (Labeled description) next.all
-            , only = List.map (Labeled description) next.only
-            , skipped = List.map (Labeled description) next.skipped
-            }
+            -- This fixes https://github.com/elm-community/elm-test/issues/192
+            -- The first time we hit a Labeled, we want to use the hash of
+            -- that label, along with the original seed, as our starting
+            -- point for distribution. Repeating this process more than
+            -- once would be a waste.
+            if hashed then
+                let
+                    next =
+                        distributeSeedsHelp True runs seed subTest
+                in
+                { seed = next.seed
+                , all = List.map (Labeled description) next.all
+                , only = List.map (Labeled description) next.only
+                , skipped = List.map (Labeled description) next.skipped
+                }
+            else
+                let
+                    intFromSeed =
+                        -- At this point, this seed will be the original
+                        -- one passed into distributeSeeds. We know this
+                        -- because the only other branch that does a
+                        -- Random.step on that seed is the Internal.Test
+                        -- branch, and you can't have a Labeled inside a
+                        -- Test, so that couldn't have come up yet.
+                        seed
+                            -- Convert the Seed back to an Int
+                            |> Random.step (Random.int 0 Random.maxInt)
+                            |> Tuple.first
+
+                    hashedSeed =
+                        -- Incorporate the originally passed-in seed
+                        (toString intFromSeed ++ description)
+                            -- Hash from String to Int
+                            |> FNV.hashString
+                            -- Convert Int back to Seed
+                            |> Random.initialSeed
```

Comment:
You might end up losing a bit of entropy by xor'ing multiple things together like this, but I think we're fine in this case. If we want to be safer, we could add a large prime and mod 32; see [source code of FNV](https://github.com/Skinney/fnv/blob/1.0.4/src/FNV.elm). This effectively just extends the FNV algorithm by one step.